### PR TITLE
Gulp release workflow

### PR DESCRIPTION
**Pull request by nicolaisueper**
This PR will add a release task which: 
- bumps the version number
- creates a release tag and generates a changelog using conventional-changelog using the angular preset
- adds all untracked files to git
- does a pre-release commit

Beside this I also add minimist for parsing our command line args. This increases the readability but we're restricted to double-dashed-args now.

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -220,5 +229,44 @@ export const build = gulp.series(
...
+function createChangelog() {
+	return gulp.src('./CHANGELOG.md', {buffer: false})
+		.pipe(changelog({preset: 'angular'}))
+		.pipe(gulp.dest('./'));
+}
+
+function commitChanges() {
+	return gulp.src('.')
+		.pipe(git.add())
+		.pipe(git.commit('[Pre-Release] Bumped version number'));
```
Comment (mischah) [inline gulpfile.babel.js]: Hard coded commit message?
I would prefer having just the version number within the commit message.

Comment (nicolaisueper) [inline gulpfile.babel.js]: Just 'x.x.x'?

Comment (mischah) [inline gulpfile.babel.js]: Or "Release v4.0.0". That's what we used to generate before. But I don't care. It's up to you.

**Comment by mischah** [comment]
Ready to merge :shipit: 

**💥 Breaking change: 💥**
Changed command line flag for production build from `-prod` to `--production`.
But also introduced an alias with `-P` 😘 

Simplified flags for release tag:
```
'gulp release --bump major': ' major release (1.0.0). See http://semver.org',
'gulp release --bump minor': ' minor release (0.1.0). See http://semver.org',
'gulp release --bump patch': ' patch release (0.0.1). See http://semver.org',
'gulp release -B major|minor|patch': ' alias to --bump'
```

**Pull request merged**